apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/SpansResourceTest.java

```diff
@@ -3412,6 +3419,41 @@ void batch__whenSendingMultipleSpansWithNoId__thenReturnNoContent() {
             batchCreateAndAssert(expectedSpans, API_KEY, TEST_WORKSPACE);
         }
 
+        @Test
+        void batch__whenCreateSpansUsageWithNullValue__thenReturnNoContent() throws JsonProcessingException {
+
+            Map<String, Integer> usage = new LinkedHashMap<>() {{
+                put("firstKey", 10);
+                put("secondKey", null);
+            }};
+
+            var expectedSpans = PodamFactoryUtils.manufacturePojoList(podamFactory, Span.class).stream()
+                    .map(trace -> trace.toBuilder()
+                            .usage(usage)
+                            .build())
+                    .toList();
+
+            var spanBatch = new SpanBatch(expectedSpans);
+
+            ObjectMapper mapper = new ObjectMapper()
+                    .setPropertyNamingStrategy(PropertyNamingStrategies.SnakeCaseStrategy.INSTANCE)
+                    .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+                    .registerModule(new JavaTimeModule().addDeserializer(BigDecimal.class, new JsonBigDecimalDeserializer()));
+
+            String body = mapper.writeValueAsString(spanBatch);
+
+            try (var actualResponse = client.target(URL_TEMPLATE.formatted(baseURI))
+                    .path("batch")
+                    .request()
+                    .header(HttpHeaders.AUTHORIZATION, API_KEY)
+                    .header(WORKSPACE_HEADER, TEST_WORKSPACE)
+                    .post(Entity.json(JsonUtils.getJsonNodeFromString(body)))) {
+
+                assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(204);
+                assertThat(actualResponse.hasEntity()).isFalse();
+            }
+        }
```

Comment:
There is no need for this new ObjectMapper. Also, we need to assert the response to see if the retrieve is as it's supposed to be.

```

```java
void batch__whenCreateSpansUsageWithNullValue__thenReturnNoContent() {

            String projectName = UUID.randomUUID().toString();

            Map<String, Integer> usage = new LinkedHashMap<>() {{
                put("firstKey", 10);
            }};

            var expectedSpans = PodamFactoryUtils.manufacturePojoList(podamFactory, Span.class).stream()
                    .map(span -> span.toBuilder()
                            .usage(usage)
                            .projectName(projectName)
                            .parentSpanId(null)
                            .feedbackScores(null)
                            .build())
                    .toList();

            var spanBatch = new SpanBatch(expectedSpans);

            JsonNode body = JsonUtils.readTree(spanBatch);

            body.get("spans").forEach(span -> {
                var usageNode = span.get("usage");

                if (usageNode instanceof ObjectNode) {
                    ((ObjectNode) usageNode).set("secondKey", NullNode.getInstance());
                }
            });

            try (var actualResponse = client.target(URL_TEMPLATE.formatted(baseURI))
                    .path("batch")
                    .request()
                    .header(HttpHeaders.AUTHORIZATION, API_KEY)
                    .header(WORKSPACE_HEADER, TEST_WORKSPACE)
                    .post(Entity.json(body))) {

                assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(204);
                assertThat(actualResponse.hasEntity()).isFalse();
            }

            getAndAssertPage(TEST_WORKSPACE, projectName, List.of(), List.of(), expectedSpans.reversed(), List.of(),
                    API_KEY);
        }

Reply:
Done